### PR TITLE
Polishing Gradle HTML UI

### PR DESCRIPTION
**Pull request by dukescript**
The HTML UI wizard needs some improvements to be considered stable.

- people reported problems with `app` subproject - let's comment it out for forthcoming release
- let's add a test to verify the generated project really builds - needs API for synchronous build execution
- using "Run" doesn't work first time, only second time

CC @lkishalmi , @neilcsmith-net

### CODE REVIEW

**Comment by lkishalmi** [comment]
"needs API for synchronous build execution" what doe that mean?

File: groovy/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java [hunk 1]
```diff
@@ -250,6 +251,9 @@ private static void invokeProjectAction(final Project project, final ActionMappi
                     prj.reloadProject(true, maxQualily, reloadArgs);
                 }
                 project.getLookup().lookup(AfterBuildActionHook.class).afterAction(action, outerCtx, task.result(), out1);
+                for (TaskListener l : context.lookupAll(TaskListener.class)) {
+                    l.taskFinished(t);
```
Comment (dukescript) [inline groovy/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java]: This is the extension created for synchronous execution of the `BUILD` and `RUN` actions in the test, @lkishalmi.

File: groovy/gradle.htmlui/test/unit/src/org/netbeans/modules/gradle/htmlui/CreateArchetypeTest.java [hunk 2]
```diff
@@ -78,11 +98,55 @@ public void testCreateFromArchetype() throws Exception {
...
+        assertFile("Main script created", dest, "web", "build", "web", "bck2brwsr.js");
+    }
+
+    protected void invokeCommand(ActionProvider actions, String cmd, Project prj) throws IllegalArgumentException, InterruptedException {
+        CountDownLatch waiter = new CountDownLatch(1);
+        TaskListener notifier = (t) -> {
+            waiter.countDown();
+        };
+        actions.invokeAction(cmd, Lookups.fixed(prj, notifier));
+        waiter.await();
```
Comment (dukescript) [inline groovy/gradle.htmlui/test/unit/src/org/netbeans/modules/gradle/htmlui/CreateArchetypeTest.java]: The `invokeCommand` has to wait till the `cmd` action is finished, @lkishalmi.

File: groovy/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java (hunk 1 above)
Comment (lkishalmi) [reply groovy/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java]: Well, I would not leak the task out of this context. As far as I see registering an AfterBuildActionHook could do the same as this task listener magic.

Comment (dukescript) [reply groovy/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java]: Rewritten to `AfterBuildActionHook` in cee89d4 - but it still needs a change in `ActionProviderImpl`. The previous code only searched `project.getLookup()`...

Comment (lkishalmi) [reply groovy/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java]: That's actually a nice enhancement. Thank you for that!

File: groovy/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java [hunk 3]
```diff
@@ -251,8 +250,8 @@ private static void invokeProjectAction(final Project project, final ActionMappi
                     prj.reloadProject(true, maxQualily, reloadArgs);
                 }
                 project.getLookup().lookup(AfterBuildActionHook.class).afterAction(action, outerCtx, task.result(), out1);
-                for (TaskListener l : context.lookupAll(TaskListener.class)) {
-                    l.taskFinished(t);
+                for (AfterBuildActionHook l : context.lookupAll(AfterBuildActionHook.class)) {
+                    l.afterAction(action, outerCtx, task.result(), out1);
                 }
```
Comment (lkishalmi) [inline groovy/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java]: That's actually a nice enhancement. Thank you for that!

**Pull request merged**